customdomain-terraform/main.tf

```diff
@@ -0,0 +1,59 @@
+variable "aws_account_role_arn" {
+  type = "string"
+}
+
+terraform {
+  backend "s3" {}
+}
+
+provider "aws" {
+  region = "eu-west-2"
+
+  assume_role {
+    role_arn = var.aws_account_role_arn
+  }
+}
+
+resource "aws_route53_zone" "eidas_zone" {
+  name = "eidas.signin.service.gov.uk"
+}
+
+module "staging_proxy_node" {
+  source          = "./modules/customdomain"
+  zone_id         = aws_route53_zone.eidas_zone.zone_id
+  govuk_domain    = "proxy-node.test.eidas.signin.service.gov.uk"
+  govsvcuk_domain = "proxy-node.test.verify-eidas-proxy-node-build.london.verify.govsvc.uk"
+}
+
+module "integration_proxy_node" {
+  source          = "./modules/customdomain"
+  zone_id         = aws_route53_zone.eidas_zone.zone_id
+  govuk_domain    = "proxy-node.integration.eidas.signin.service.gov.uk"
+  govsvcuk_domain = "proxy-node.integration.verify-eidas-proxy-node-deploy.london.verify.govsvc.uk"
+}
+
+module "prod_proxy_node" {
+  source          = "./modules/customdomain"
+  zone_id         = aws_route53_zone.eidas_zone.zone_id
+  govuk_domain    = "proxy-node.eidas.signin.service.gov.uk"
+  govsvcuk_domain = "proxy-node.production.verify-eidas-proxy-node-deploy.london.verify.govsvc.uk"
+  # TODO: this will need to change per https://www.terraform.io/docs/providers/aws/r/acm_certificate.html#importing-an-existing-certificate when we get a QWAC cert issued
+}
+
+module "staging_stub_connector" {
+  source          = "./modules/customdomain"
+  zone_id         = aws_route53_zone.eidas_zone.zone_id
+  govuk_domain    = "stub-connector.test.eidas.signin.service.gov.uk"
+  govsvcuk_domain = "stub-connector.test.verify-eidas-proxy-node-build.london.verify.govsvc.uk"
+}
+
+module "integration_stub_connector" {
+  source          = "./modules/customdomain"
+  zone_id         = aws_route53_zone.eidas_zone.zone_id
+  govuk_domain    = "stub-connector.integration.eidas.signin.service.gov.uk"
+  govsvcuk_domain = "stub-connector.integration.verify-eidas-proxy-node-deploy.london.verify.govsvc.uk"
+}
+
+output "eidas_zone_id" {
```

Comment:
Is it worth including the `name_servers` for the zone as output as well?

Reply:
Oh, oops, that's what we should be putting in the output instead of the zone ID. Well spotted